### ISSUE TITLE
lio_read should return the number of bytes actually read

### ISSUE THREAD

**Issue by PerilousApricot** [accre/lstore#126]
Currently, `lio_read` just kicks back the same size that the user passed in, even if that read would go beyond the end of the file:

https://github.com/accre/lstore/blob/master/src/lio/lio_core_io.c#L963

That means there's no way to know "where" the EOF was hit, just that the next time you read you'll get zero bytes

**Comment by tacketar** [comment]
Good catch.  It's looks to be a pretty easy fix.  I'll take care of it 
next week when I write the example GOP and open/write/read/close 
examples.  It can be one of the examples. It's complicated by the read 
ahead cache buffering but it looks like _gop_lio_read() calculates the 
actual number of bytes dumped in the buffer excluding any read ahead.  
That just needs to get passed back and returned as the actual number of 
bytes instead of what the user passes in.

**Comment by PerilousApricot** [comment]
Prolly worth checking the write variant. Gridftp plugin is blocking on
this, FWIW

## 

It's dark in this basement.

**Comment by tacketar** [comment]
Write looks fine.

**Comment by PerilousApricot** [comment]
Any chance you could pop out a quick patch to the read path? I can't find a
good way around it for the gridftp plugin.

**Comment by tacketar** [comment]
Sure.  I'll knock it out this afternoon.

**Comment by PerilousApricot** [comment]
cheers.
